lib/Frontend/CompilerInvocation.cpp

```diff
@@ -679,6 +680,54 @@ void LangOptions::setCxxInteropFromArgs(ArgList &Args,
       cxxInteropCompatVersion =
           validateCxxInteropCompatibilityMode("swift-5.9").second;
   }
+
+  if (Arg *A = Args.getLastArg(options::OPT_formal_cxx_interoperability_mode)) {
```

Comment:
Could the user use this flag from the command line (as opposed to the swiftinterface file)? If they do that should that be diagnosed?

Reply:
Do you mean, raise a warning if they specify the flag when it's not a textual interface?

Reply:
Yup.

Reply:
I'm looking at this, and unless I'm missing something, it doesn't seem like there's a straightforward way to determine what we're looking at (textual interface vs otherwise).

Now I'm thinking that maybe it's not very useful to warn the user about this. It's a hidden frontend-only flag anyway, so I'm going to assume that a user that goes out of their way to pass this flag manually will hopefully know what they're getting themselves into (which isn't very much, for now).

Reply:
I don't think it's super important to diagnose this. Frontend flags are implementation details that users are never expected to interact with directly. If it were a driver flag it would be a different story.

Reply:
Makes sense, thanks!